tock/utilization/views.py

```diff
@@ -0,0 +1,160 @@
+import datetime
+
+from django.db.models import Sum, Prefetch
+from django.views.generic import ListView
+from django.contrib.auth.models import User
+
+from hours.models import Timecard, TimecardObject, ReportingPeriod
+from employees.models import UserData
+
+from .utils import get_fy_first_day, get_dates, calculate_utilization
+
+class GroupUtilizationView(ListView):
+    template_name = 'utilization/group_utilization.html'
+
+    def get_queryset(self):
+        """Gets submitted timecards limited to the reporting periods in
+        question."""
+
+        """Although recent_rps is set to last four reporting periods, could
+        accept a form response that allows the user or app to dynamically
+        customize number of periods to include in the queryset."""
+
+        available_periods = len(ReportingPeriod.objects.all())
+        requested_periods = 4
+
+        if available_periods >= requested_periods:
+            recent_rps = get_dates(requested_periods)
+        else:
+            recent_rps = get_dates(available_periods)
+
+        billable_staff = User.objects.filter(
+            user_data__is_billable=True,
+            user_data__current_employee=True
+        ).prefetch_related('user_data')
+
+        for staffer in billable_staff:
+            staffer.unit = staffer.user_data.unit
+
+            """Create smallest possible TimecardObject queryset based on the
+            earliest_date value returned by get_dates(). Also prefetches the
+            related user and accounting code for later use."""
+            max_tos = TimecardObject.objects.filter(
+                timecard__user=staffer,
+                submitted=True,
+                timecard__reporting_period__start_date__gte=recent_rps[3]
+            ).prefetch_related(
+                'timecard__user'
+            ).prefetch_related(
+                'project__accounting_code'
+            )
```

Comment:
These two `prefetch_related` called could be combined into one:

`prefetch_related('timecard__user', 'project__accounting_code')`

The order does matter, but I think that's more in cases where you have one field referencing another.  More information can be found here:  https://docs.djangoproject.com/en/1.10/ref/models/querysets/#prefetch-related